public/app/features/plugins/sandbox/sandbox_plugin_loader.ts

```diff
@@ -0,0 +1,180 @@
+      // trace that is difficult to read due to all the sandboxing
+      // layers.
+      instrumentation: {
+        // near-membrane concept of "activity" is something that happens inside
+        // the plugin instrumentation
+        startActivity() {
+          return {
+            stop: () => {},
+            error: getActivityErrorHandler(meta.id),
+          };
+        },
+        log: () => {},
+        error: () => {},
+      },
+    });
+
+    // fetch and evalute the plugin code inside the sandbox
+    try {
+      let pluginCode = await getPluginCode(meta.module);
+      if (isDevMode) {
+        pluginCode = patchPluginSourceMap(meta, pluginCode);
+      }
+
+      // runs the code inside the sandbox environment
+      // this evaluate will eventually run the `define` function inside
+      // of endowments.
+      sandboxEnvironment.evaluate(pluginCode);
+    } catch (e) {
+      reject(new Error(`Could not execute plugin ${meta.id}: ` + e));
+    }
+  });
+}
+
+async function getPluginCode(modulePath: string) {
+  const response = await fetch('public/' + modulePath + '.js');
+  return await response.text();
+}
+
+function getActivityErrorHandler(pluginId: string) {
+  return async function error(proxyError?: Error & { sandboxError?: boolean }) {
+    if (!proxyError) {
+      return;
+    }
+    // flag this error as a sandbox error
+    proxyError.sandboxError = true;
+
+    //  create a new error to unwrap it from the proxy
+    const newError = new Error(proxyError.message.toString());
+    newError.name = proxyError.name.toString();
+    newError.stack = proxyError.stack || '';
+
+    // If you are seeing this is because
+    // the plugin is throwing an error
+    // and it is not being caught by the plugin code
+    // This is a sandbox wrapper error.
+    // and not the real error
+    console.log(`[sandbox] Error from plugin ${pluginId}`);
+    console.error(newError);
+  };
+}
+
+/**
+ * Patches the plugin's module.js source code references to sourcemaps to include the full url
+ * of the module.js file instead of the regular relative reference.
+ *
+ * Because the plugin module.js code is loaded via fetch and then "eval" as a string
+ * it can't find the references to the module.js.map directly and we need to patch it
+ * to point to the correct location
+ */
+function patchPluginSourceMap(meta: PluginMeta, pluginCode: string): string {
+  // skips inlined and files without source maps
+  if (pluginCode.includes('//# sourceMappingURL=module.js.map')) {
+    let replaceWith = '';
+    // make sure we don't add the sourceURL twice
+    if (!pluginCode.includes('//# sourceURL') || !pluginCode.includes('//@ sourceUrl')) {
+      replaceWith += `//# sourceURL=module.js\n`;
+    }
+    // modify the source map url to point to the correct location
+    const sourceCodeMapUrl = `/public/${meta.module}.js.map`;
+    replaceWith += `//# sourceURL=module.js\n//# sourceMappingURL=${sourceCodeMapUrl}`;
```

Comment:
Are you sure we need the `//# sourceuRL=module.js` part on this line, too? Wouldn't this mean that we add `sourceURL` regardless it was already there? Maybe I am missing something.

Reply:
In previous lines that was verified (mentioned in the comments)

Reply:
Unfortunately I don't think that's how it works. 

![Screenshot 2023-05-30 at 14 01 31](https://github.com/grafana/grafana/assets/9974811/19248d65-00d1-438e-97a3-d0e8f2ead8a4)

![Screenshot 2023-05-30 at 13 59 46](https://github.com/grafana/grafana/assets/9974811/6e597ce4-79aa-40d2-8c93-5b9cd0e76e35)

Reply:
It should probably be this?

```suggestion
    replaceWith += `//# sourceMappingURL=${sourceCodeMapUrl}`;
```

Reply:
Thanks for double checking this. Indeed the condition before this one was making sure the `sourceUrl` part was present and adding it if not but then here I'll add it once again.

I removed the part you suggested in this line.